src/palette/internal/palette.cpp

```diff
@@ -231,8 +230,11 @@ int Palette::indexOfCell(const PaletteCell& cell, bool matchName) const
             continue;
         }
 
-        const EngravingItem* exElement = localCell.element.get();
-        if (exElement && !isEquivalent(*exElement, *el)) {
+        if (cell.untranslatedElement) {
```

Comment:
Why do we need two instances of the same element? Or what is the difference between cell.untranslatedElement and cell.element?

Reply:
This is something specific for `TextBase` derived elements. For these elements, `element` is (as always) the element that is drawn in the palette and that gets added to the score, and `untranslatedElement` is just what the name might suggest: a copy of `element`, but with the untranslated text. 

If the cell is made translatable (i.e. `untranslatedElement` is not null), both `element` and `untranslatedElement` are written to disk when the PaletteCell is written to disk. When the user changes the language and restarts MuseScore, when reading the PaletteCell, we look at the text of `untranslatedElement`, and guess the translation context using the `translationContext` method. That way, we can reconstruct the translated `element`, so that `element` will always use the language that the user has currently selected. 

(Note: this is historical code. Or course, instead of storing a whole copy of the element, we could also just store the untranslated text as a string (or preferably even `TranslatableString`, once we have a way to write `TranslatableStrings` to XML). However, that solution would cause difficulties if we ever want to store an element in Palettes that contains _two_ translatable strings (for example, a text line with begin text and end text). So I propose to leave it as it is for now.)